Check for backtrace symbol and link to execinfo if needed

DanySpin97 opened the pull request:
I assume that backtrace symbol is needed for libwinpr (as my build failed to link without it), so cmake now errors when it cannot find `backtrace` function.

[libexecinfo](https://pkgs.alpinelinux.org/package/edge/main/x86/libexecinfo) is a library that provides backtrace header and symbol on systems that don't have it (like musl based ones). This PR should be safe to add for glibc users.

akallabeth commented on winpr/libwinpr/CMakeLists.txt:
```diff
@@ -87,6 +87,17 @@ if (TIMER_CREATE AND TIMER_DELETE AND TIMER_SETTIME AND TIMER_GETTIME)
+    set(CMAKE_REQUIRED_LIBRARIES execinfo)
+    check_function_exists(backtrace EXECINFO)
+    if (NOT EXECINFO)
+        message(FATAL_ERROR "Could not find backtrace function")
```
Pleas no fatal errors, linux is not the only platform we support.

DanySpin97 replied:
Fixed.

akallabeth commented:
@freerdp-bot test

akallabeth reviewed (changes requested):
Patch now looks good, but there is already a check in 
`winpr/libwinpr/utils/CMakeLists.txt`
- so better move it there. (sorry, did not check that before)

DanySpin97 commented:
> Patch now looks good, but there is already a check in
> `winpr/libwinpr/utils/CMakeLists.txt`
> 
>     * so better move it there. (sorry, did not check that before)

Oh, I've completely missed that. I have moved the check there and restricted it to unix systems only. I've also updated the commit message.

akallabeth commented:
@freerdp-bot test
